markdown/events/2021/bytesize-3-code-structure.md

```diff
@@ -30,3 +30,109 @@ She will talk through the different files found in the nf-core template and desc
+**This text has been edited to make it more suitable for reading**
+
+Welcome everybody, I would like to do a pipeline code walkthrough today of the DSL1 pipeline template, and also provide a preview of the DSL2 nf-core pipeline template.
+
+All the nf-core pipelines are created by using a pipeline template that contains most of the functions that the nf-core pipelines have. This makes it easy to automatically update the nf-core pipeline code with the newest developments.
+
+To generate a new nf-core pipeline, you first need to install the nf-core tools with pip or Conda and then run the command nf-core create.
+
+Doing this, automatically creates a pipeline template, and you will first be prompted for some pipeline information: like what you would like to name your new pipeline, a short description of the pipeline, who the main authors of this pipeline are, and in less than a second, you can have a pipeline structure ready!
+
+However, as you might see, the first pipeline template already contains six directories and 26 files that are visible. Now this together with hidden files can be around 40 files, so it can be a bit daunting to get started with developing or contributing to nf-core pipelines.
+
+I'm going to demonstrate the function of all of the individual files inside the template, and talk about why we have them there.
+
+We will start easy.
+
+First of all, the most important file of the whole template is the main.nf file, and this contains the main Nextflow pipeline code. No further introduction is needed for this one.
+
+Additionally we have other files, and different Github repositories for different projects such as the changelog/markdown file. This one contains a summary of the changes that have been introduced into the pipeline between the different releases.
+
+We also have a code of conduct file that describes the expected behavior of people that want to contribute to the community and to the pipelines.
+
+We also have the license file, which describes the MIT license that all of the nf-core pipelines adhere to or apply, and a readme markdown file.
+
+The readme markdown file contains the easiest pipeline recommendations, such as how to get started to launch the pipeline.
+
+What are pipelines actually? They enable the execution of different tools one after another, so it's pretty important for any pipeline to have proper software packaging possibilities. This is natively supported by Nextflow; Nextflow natively supports the execution of pipelines using many different container engines and also Conda.
+
+We support this also of course in all of the nf-core pipelines.
+
+So how is software packaging defined?
+
+As part of the nf-core pipelines, we have mainly two files: the Dockerfile and the environment.yml. The environment.yml is a Conda environment file that lists all
+of the software dependencies of the pipeline and their versions. It can be used to just execute the pipeline by using Conda.
```

Comment:
```suggestion
As part of the nf-core pipelines, we have mainly two files: the Dockerfile and the environment.yml. The environment.yml is a Conda environment file that lists all of the software dependencies of the pipeline and their versions. It can be used to just execute the pipeline by using Conda.
```